measurement/admin.py

```diff
@@ -1,23 +1,67 @@
 from django.contrib import admin
+from django.core.exceptions import PermissionDenied
+from guardian.admin import GuardedModelAdmin
+from guardian.shortcuts import get_perms
 
-from .models import Measurement, Report
+from measurement.models import Measurement, Report
+from station.models import Station
 
 admin.site.site_header = "Paricia Administration - Measurements"
 
 
-class MeasurementBaseAdmin(admin.ModelAdmin):
+class MeasurementBaseAdmin(GuardedModelAdmin):
     list_display = ["id", "station", "variable", "maximum", "minimum"]
     list_filter = ["station", "variable"]
 
+    def has_add_permission(self, request):
+        """Allow all authenticated users to add objects."""
+        return request.user.is_authenticated
+
+    def has_change_permission(self, request, obj=None):
+        """Check if the user has the correct permission to change the object."""
+        if obj is not None:
+            return "change_station" in get_perms(request.user, obj.station)
+        return True
+
+    def has_delete_permission(self, request, obj=None):
+        """Check if the user has the correct permission to delete the object."""
+        if obj is not None:
+            return "delete_station" in get_perms(request.user, obj.station)
+        return True
+
+    def has_view_permission(self, request, obj=None):
+        """Check if the user has the correct permission to view the object."""
+        if obj is not None:
+            return "view_measurements" in get_perms(request.user, obj.station)
+        return True
+
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
+        """Limit the list of stations available based on permssions."""
+        if db_field.name == "station":
+            kwargs["queryset"] = Station.objects.filter(owner=request.user)
+        return super().formfield_for_foreignkey(db_field, request, **kwargs)
```

Comment:
I'm not entirely sure what this one does. Are we filtering the list of measurements that are available in the Admin based on the stations the user is owner? And what about public measurements?

Reply:
This filters the list of stations available to use when creating or modifying measurements in the admin page, i.e. to prevent people from creating measurements for stations they do not own.